python/python/Ice/ObjectPrx.py

```diff
@@ -11,10 +11,19 @@
 from .Object import Object
 
 if TYPE_CHECKING:
-    import Ice
+    from typing import Type, TypeVar
 
+    from .Communicator import Communicator
+    from .EncodingVersion import EncodingVersion
+    from .EndpointSelectionType import EndpointSelectionType
+    from .Identity import Identity
+    from .Locator import LocatorPrx
+    from .Router import RouterPrx
 
-def uncheckedCast(type, proxy, facet=None):
+    T = TypeVar("T", bound=ObjectPrx)
+
+
+def uncheckedCast(type: Type[T], proxy: ObjectPrx, facet: str | None = None) -> T | None:
```

Comment:
proxy can be None according to the implementation. 

Can we use overloads? either pass None and returns None, or pass a proxy and returns a proxy of type T